algorithmic_efficiency/workloads/wmt/input_pipeline.py

```diff
@@ -7,9 +7,11 @@
 import tensorflow_datasets as tfds
 
 from algorithmic_efficiency import data_utils
+from algorithmic_efficiency.pytorch_utils import pytorch_setup
 from algorithmic_efficiency.workloads.wmt import tokenizer
 
-AUTOTUNE = tf.data.AUTOTUNE
+RANK = pytorch_setup()[1]
+AUTOTUNE = tf.data.AUTOTUNE if RANK == 0 else None
```

Comment:
is this saying, set to None if on the first process? can you add a quick comment stating that?

Reply:
It should only use `tf.data.AUTOTUNE` in the first process (rank 0), to avoid multithreading in all other processes. If PyTorch DDP is not used, e.g. when Jax is being used, `RANK` will be `0`, so `AUTOTUNE` will also be set correctly. I have added a clarifying comment.